FirebasePerformance/Sources/AppActivity/FPRSessionManager.m

```diff
@@ -104,8 +105,12 @@ - (void)updateSessionId:(NSNotification *)notification {
   FPRSessionDetails *sessionInfo = [[FPRSessionDetails alloc] initWithSessionId:sessionIdString
                                                                         options:sessionOptions];
   self.sessionDetails = sessionInfo;
+  NSMutableDictionary<NSString *, FPRSessionDetails *> *userInfo =
+      [[NSMutableDictionary alloc] init];
+  [userInfo setObject:sessionInfo forKey:kFPRSessionIdNotificationKey];
   [self.sessionNotificationCenter postNotificationName:kFPRSessionIdUpdatedNotification
-                                                object:self];
+                                                object:self
+                                              userInfo:[userInfo copy]];
```

Comment:
Do we need to copy this? Would this not have a similar race condition?

Reply:
This is the userInfo object copy and this is a local variable. So no real possibility of a race condition here. The FPRSessionDetails object is already help by the dictionary, so that is not going out of memory to cause a crash.